components/IncomeExpenseView.js

```diff
@@ -0,0 +1,59 @@
+import { STATE } from "@/constants/state";
+import styled from "styled-components";
+import { useState } from "react";
+
+export default function IncomeExpenseView({
+  filteredTransactions,
+  sumIncome,
+  sumExpense,
+  sumTotal,
+  onFilter,
+}) {
+  const [filterType, setFilterType] = useState(STATE.ALL);
+
+  //Change radio button
+  const handleFilterType = (event) => {
+    const filterValue = event.target.value;
+    setFilterType(filterValue);
+    if (onFilter) {
+      onFilter(filterValue);
+    }
+  };
```

Comment:
Let me be more clear about this issue (what I ment in the last code review):
- This `filterType` useState is not necessary and it is confusing, too. Because you already handle the filterType state in 'pages/index.js' and that is fine. So instead of defining it here again you should pass it as a prop. 
- Inside the handleFilterType function you can simple call `onFilter(filterValue)`. You can omit `setFilterType(...)` and the `if (onFilter)`, too.

(If you need more infos, please reach out to me)